### PR TITLE
[node] add util.formatWithOptions

### LINKED ISSUE #28221: [@types/node] v10.7.1 don`t have definition of "util.formatWithOptions"
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/node` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @tdreyno

missing definition of [util.formatWithOptions](https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_formatwithoptions_inspectoptions_format_args)

### PR DESCRIPTION
**Pull request by Flarna**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [y] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_formatwithoptions_inspectoptions_format_args
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

fixes #28221 
fyi @mx601595686

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -6331,6 +6331,7 @@ declare module "stream" {
 declare module "util" {
     export interface InspectOptions extends NodeJS.InspectOptions { }
     export function format(format: any, ...param: any[]): string;
+    export function formatWithOptions(inspectOptions: InspectOptions, format: any, ...param: any[]): string;
```
Comment (eps1lon) [inline types/node/index.d.ts]: While you can pass `any` since node seems to convert it to string anyway the docs specifically mention `string`. Maybe this is out of scope for this PR because this is also typed in `format`. Is there a reason to use `any`?

Comment (Flarna) [reply types/node/index.d.ts]: Thanks!
No special reason. I just did a simple combine of `inspect` and `format` and forgot to take a closer look to them first. I corrected `formatWithOptions` but keep format as it is for now.

**Review by eps1lon — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]
